### DISCUSSION TITLE
Running on a remote machine

### DISCUSSION THREAD

**Discussion by antopolskiy** [code-kern-ai/refinery#125] [Q&A]
Hi! I'd really like to try out Refinery for my pet project, but on my machine Docker is not running some of the images correctly (is it a known issue on M1 chips).

I tried to set up Refinery on a virtual machine on AWS EC2, and discovered that I am unable to connect to the UI, getting `{"error":{"code":404,"status":"Not Found","message":"Requested url does not match any rules"}}`, but MailHog works ok so I know it is not an issue of network security. 

Do I understand correctly that this is intentionally forbidden through oathkeeper access policy and the UI is supposed to by accessed only on the host which is running `docker-compose`?

Is there any way for me to try Refinery for my project if my machine is unable to run it in Docker?

**Comment by jhoetter** [comment]
Hi, thanks for your message!

Currently, refinery is meant to run on your localhost. We're more than happy to help you there, most of our devs have M1 chips, so if you tell us the issue we can maybe help you there :)

If it helps for your pet project, and you really can't run refinery on your local machine, we're also happy to give you access to the managed cloud for 14 days for free. Is this enough time for the pet project?

**Reply by antopolskiy** [reply]
Hi @jhoetter!

The issue in my case, as far as I can tell, is the following (I had it with other docker images, such as the official Metabase image): some of the docker images have to run in emulation mode, and their performance suffers significantly to the point where they tend to freeze, leaving parts of the software not functioning. This doesn't happed every time, but with some probability these images crash or freeze.

<img width="829" alt="image" src="https://user-images.githubusercontent.com/23148726/194123846-c7aa081e-4c56-4730-87ba-eebeadeaceb8.png">

I am running Docker Desktop v4.12.0 on Macbook Air M1 16GB, and the latest version of Refinery.

**Reply by jhoetter** [reply]
@FelixKirsch this sounds similar to what we faced. Do you have an idea here?

**Reply by FelixKirsch** [reply]
Hi @antopolskiy,
in our current setup, some third-party containers run via emulation on M1. I have never experienced any performance issues with it. For the moment, I would ignore the warnings in docker desktop. But I'll create an issue to check if these third-party containers are also provided as arm build.
Can you open the application after starting refinery on http://localhost:4455/app/ ?

**Reply by FelixKirsch** [reply]
https://github.com/code-kern-ai/refinery/issues/126
